.buildkite/scripts/install_deps.sh

```diff
@@ -35,6 +35,10 @@ add_bin_path(){
 }
 
 with_docker() {
+    if [[ "${platform_type_lowercase}" == "darwin" ]]; then
+        echo "Skip. Not supported in Darwin."
+        return
+    fi
```

Comment:
This is not needed now, right?

Reply:
It is still needed. These function (with_docker and with_docker_compose) are called from the script `.buildkite/scripts/integrations_tests.sh`. This script is used for all the steps that could run with Linux containers or Linux VMs, so it is needed to keep these safeguards/checks.

https://github.com/elastic/elastic-package/blob/10509f6bcb3a22a2c57a353e97ea1bfa46a9608e/.buildkite/scripts/integration_tests.sh#L83-L88

Reply:
Maybe I am missing something, but do we want these scripts to continue if docker installation cannot be performed?

> This script is used for all the steps that could run with Linux containers or Linux VMs, so it is needed to keep these safeguards/checks.

Is it run in darwin?

Reply:
> > This script is used for all the steps that could run with Linux containers or Linux VMs, so it is needed to keep these safeguards/checks.
> 
> Is it run in darwin?

Ok, I see it is run with `-t test-just-build-zip`, could we avoid calling `with_docker` and `with_docker_compose_plugin` when the target is `test-just-build-zip`? The same way as we are avoiding it when running for serverless.

Reply:
Exactly, for now just Macos ARM is used for that specific make target.
It could be added as you mentioned that condition to not call those functions if that target is the one to be executed.

I'll give it a try :+1:

Reply:
Applied the suggestion here https://github.com/elastic/elastic-package/pull/2030/commits/2e7220d9bdb73d6b8f403875ceed6549e09a1efd